src/queries/comment/createdAt.ts

```diff
@@ -0,0 +1,21 @@
+import type { GQLCommentResolvers } from 'definitions'
+
+import { COMMENT_STATE } from 'common/enums'
+
+const resolver: GQLCommentResolvers['createdAt'] = (
+  { createdAt, state },
+  _,
+  { viewer }
+) => {
+  const isActive = state === COMMENT_STATE.active
+  const isCollapsed = state === COMMENT_STATE.collapsed
+  const isAdmin = viewer.hasRole('admin')
+  if (isActive || isCollapsed || isAdmin) {
+    return createdAt
+  } else {
+    // invalid date
+    return new Date(0)
```

Comment:
should return `null` instead?

Reply:
`createdAt: DateTime!`
can not return null here without update schema